crates/sources/src/loader.rs

```diff
@@ -68,6 +68,48 @@ pub struct Loader<F: Fetcher> {
     fetcher: F,
 }
 
+/// Parses a flow catalog specification **without** perfoming any sort of resolution of any other
+/// resources. This takes care of all the workarounds for dealing with `serde_json::RawValue`s, as
+/// well as resolving yaml merge keys.
+pub fn parse_catalog_spec(content: &[u8]) -> Result<models::Catalog, LoadError> {
+    let json = load_resource_content_dom(content, flow::ContentType::Catalog)?;
+    let catalog = serde_json::from_str(json.get())?;
+    Ok(catalog)
+}
+
+/// Parses `content` into a JSON DOM that can be further parsed into a Rust type.
+fn load_resource_content_dom(
+    content: &[u8],
+    content_type: flow::ContentType,
+) -> Result<Box<RawValue>, LoadError> {
+    use flow::ContentType as CT;
+
+    // These types are not documents and have a placeholder DOM.
```

Comment:
This is very likely just some nuance I don't understand, but it looks like you're replacing npm/typescript config with null, and the comment doesn't help me understand why (isn't e.g the npm config actually a document containing packages and versions to install?)

Reply:
Well maybe nevermind, this isn't new code, it was just moved. /shrug I assume it's there for a reason

Reply:
Yeah, it's existing, but it's because there's no way to directly represent a typescript module or npm package as JSON. `load_resource_content` could definitely be refactored to remove this weirdness, but I was trying to keep my changes to this crate as minimal as possible in this PR, since such a refactor is pretty unrelated